crates/vm/levm/src/opcode_handlers/arithmetic.rs

```diff
@@ -147,22 +147,28 @@ impl<'a> VM<'a> {
         let [augend, addend, modulus] = *current_call_frame.stack.pop()?;
 
         if modulus.is_zero() {
-            current_call_frame.stack.push(&[U256::zero()])?;
+            current_call_frame.stack.push1(U256::zero())?;
             return Ok(OpcodeResult::Continue { pc_increment: 1 });
         }
 
         let new_augend: U512 = augend.into();
         let new_addend: U512 = addend.into();
 
-        let sum = new_augend
-            .checked_add(new_addend)
-            .ok_or(InternalError::Overflow)?;
+        #[allow(
+            clippy::arithmetic_side_effects,
+            reason = "both values come from a u256, so the product can fit in a U512"
+        )]
+        let sum = new_augend + new_addend;
+        #[allow(
+            clippy::arithmetic_side_effects,
+            reason = "can't overflow because non-zero modulus"
```

Comment:
This case was never about overflow, but avoid division by zero. The closest one-word description I could think of is "trap":
```suggestion
            reason = "can't trap because non-zero modulus"
```